paasta_tools/cli/cmds/mark_for_deployment.py

```diff
@@ -465,28 +466,51 @@ def paasta_mark_for_deployment(args: argparse.Namespace) -> None:
     if not can_user_deploy_service(deploy_info, service):
         sys.exit(1)
 
-    deploy_process = MarkForDeploymentProcess(
-        service=service,
-        deploy_info=deploy_info,
-        deploy_group=deploy_group,
-        commit=commit,
-        old_git_sha=old_git_sha,
-        git_url=args.git_url,
-        auto_rollback=args.auto_rollback,
-        block=args.block,
-        soa_dir=args.soa_dir,
-        timeout=args.timeout,
-        warn_pct=args.warn,
-        auto_certify_delay=args.auto_certify_delay,
-        auto_abandon_delay=args.auto_abandon_delay,
-        auto_rollback_delay=args.auto_rollback_delay,
-        authors=args.authors,
-        polling_interval=args.polling_interval,
-        diagnosis_interval=args.diagnosis_interval,
-        time_before_first_diagnosis=args.time_before_first_diagnosis,
+    metrics_factory: Callable[[str], metrics_lib.BaseMetrics] = metrics_lib.NoMetrics
+    if args.block:  # only time if wait for deployment
+        metrics_factory = metrics_lib.get_metrics_interface
+    metrics = metrics_factory("paasta.mark_for_deployment")
+    deploy_timer = metrics.create_timer(
+        name="deploy_duration",
+        default_dimensions=dict(
+            paasta_service=service,
+            deploy_group=deploy_group,
+            old_version=old_git_sha,
+            new_version=commit,
```

Comment:
should we skip metric emission if `old_git_sha == commit`? that'll mess with the timings otherwise

Reply:
Good call.